src/zenml/steps/utils.py

```diff
@@ -318,6 +318,17 @@ def resolve_input_artifact(
         Returns:
             Return the output of `handle_input()` of selected materializer.
         """
+        # Make an exception for all BaseArtifact, return artifact
+        if issubclass(data_type, BaseArtifact):
+            if data_type != type(artifact):
+                logger.warning(
+                    f"You specified the data_type `{data_type}` but the actual "
```

Comment:
I'm kind of surprised our linter does not complain about using f-strings in logger calls, I seem to think it busted me for it once before.

Reply:
I think that is in case of Exceptions not logs